lib/src/main/java/org/esbtools/eventhandler/NotificationRepository.java

```diff
@@ -33,5 +33,5 @@
     List<Notification> retrieveOldestNotificationsUpTo(int maxEvents) throws Exception;
 
     // TODO: What if failed to create document events (this is definitely a possibility)
-    void confirmProcessed(Collection<Notification> notification) throws Exception;
+    void confirmProcessedNotifications(Collection<Notification> notification) throws Exception;
```

Comment:
+1 to that

I also need to add failed notifications, so maybe something like `markNotificationsAsProcessedOrFailed`. Is that too verbose? It should be the same API so it supports a bulk update of everything at once.